src/hardware/sblaster.cpp

```diff
@@ -126,6 +125,21 @@ enum class DmaMode {
 
 enum class EssType { None, Es1688 };
 
+class Dac {
+public:
+	std::optional<int> MeasureDacRateHz();
+	AudioFrame RenderFrame();
+
+private:
+	// 1% change threshold
+	static constexpr int ChangeTallyThreshold = 10;
```

Comment:
So ` RateChangeThresholdPercent` is clear, but what about this? Is this about the rate change tally... or something else?

Reply:
Thanks; it wasn't that clear. 
Renamed slightly and added more comments.

```c++
public:
	// When the DAC is in use, we run the Sound Blaster at exactly the rate
	// the DAC is being written to and generate frame by frame. To support
	// this, we need to measure the rate the DAC is being written to.
	std::optional<int> MeasureDacRateHz();
	AudioFrame RenderFrame();

private:
	// We use two criteria to monitor and decide when the rate's changed:
	// percent difference (versus current) and when the new rate persists
	// across a sequential count. These two thresholds (1% change confirmed
	// across 10 sequential changes) were selected based on inspecting
	// actual games and demos (Alone in the Dark, Overload demo, EMF demo,
	// Cronolog demo, Chickens demo). These get the timing right without
	// whip-sawing or chasing.
	//
	static constexpr float PercentDifferenceThreshold = 0.01f;
	static constexpr int SequentialChangesThreshold   = 10;
```

Reply:
Great stuff, man! 🚀